components/utils/renderIcon.tsx

```
@@ -0,0 +1,66 @@
+
+        return () => {
+            active = false;
+        };
+    }, [icon]);
+
+    if (!icon) return null;
+
+    // 1. 内联 svg
+    if (icon.trim().startsWith('<svg')) {
+        return <span dangerouslySetInnerHTML={{ __html: icon }} />;
+    }
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)